core/server/models/role.js

```diff
@@ -90,6 +90,15 @@ Role = ghostBookshelf.Model.extend({
             hasUserPermission = roleModelOrId && _.includes(checkAgainst, roleModelOrId.get('name'));
         }
 
+        if (action === 'assign' && loadedPermissions.apiKey) {
+            // apiKey cannot 'assign' the 'Owner' role
+            if (roleModel.get('name') === 'Owner') {
```

Comment:
rather than doing the array check stuff here - i have explicitly tested against if it is an owner role, i think that this more accurately represents the requirements and is easier to understand

Reply:
yeah i like